langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/CloneWithType.java

```diff
@@ -105,15 +106,30 @@ private static Object convert(Object value, Type targetType, List<TypeValuePair>
             if (targetType.isNilable()) {
                 return null;
             }
-            return createError(CONSTRUCT_FROM_CONVERSION_ERROR,
+            return createError(VALUE_LANG_LIB_CONVERSION_ERROR,
                     BLangExceptionHelper.getErrorMessage(RuntimeErrors.CANNOT_CONVERT_NIL, targetType));
         }
+
+        List<String> errors = new ArrayList<>();
         Set<Type> convertibleTypes;
-        convertibleTypes = TypeConverter.getConvertibleTypes(value, targetType);
+        convertibleTypes = TypeConverter.getConvertibleTypes(value, targetType, null, false, errors);
 
         Type sourceType = TypeChecker.getType(value);
         if (convertibleTypes.isEmpty()) {
-            throw createConversionError(value, targetType);
+            if (errors.isEmpty()) {
+                throw createConversionError(value, targetType);
+            } else {
+                StringBuilder errorMsg = new StringBuilder();
+                byte errorCount = 0;
+                for (String error : errors) {
+                    errorMsg.append("\n\t\t").append(error);
+                    errorCount++;
+                    if (errorCount == 100) {
+                        break;
```

Comment:
Shall we add `...` part in the large error messages to indicate the continuation?